odoo-openmrs/src/main/java/com/ozonehis/eip/odoo/openmrs/client/OpenmrsRestClient.java

```diff
@@ -0,0 +1,110 @@
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.ozonehis.eip.odoo.openmrs.client;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublisher;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandler;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.Base64;
+import java.util.Set;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+
+/**
+ * Submits http requests to OpenMRS.
+ */
+@Component
+public class OpenmrsRestClient {
+
+    protected static final String PATH = "/ws/rest/v1/";
+
+    protected static final BodyHandler<byte[]> BODY_HANDLER = BodyHandlers.ofByteArray();
+
+    @Value("${openmrs.baseUrl}")
+    private String baseUrl;
+
+    @Value("${openmrs.username}")
+    private String username;
+
+    @Value("${openmrs.password}")
+    private char[] password;
+
+    private HttpClient client = HttpClient.newHttpClient();
+
+    public byte[] get(String resource, String uuid) throws Exception {
+        HttpResponse<byte[]> response =
+                sendRequest(resource, uuid, null, false, Set.of(HttpStatus.OK.value(), HttpStatus.NOT_FOUND.value()));
+        if (response.statusCode() == HttpStatus.NOT_FOUND.value()) {
+            return null;
+        }
+
+        return response.body();
+    }
+
+    public byte[] createOrUpdate(String resource, String uuid, String body) throws Exception {
+        final HttpStatus expectedStatus = (uuid == null) ? HttpStatus.CREATED : HttpStatus.OK;
+        return sendRequest(resource, uuid, body, false, Set.of(expectedStatus.value()))
+                .body();
+    }
+
+    public byte[] delete(String resource, String uuid) throws Exception {
+        return sendRequest(resource, uuid, null, true, Set.of(HttpStatus.NO_CONTENT.value()))
+                .body();
+    }
+
+    public HttpResponse<byte[]> sendRequest(
+            String resource, String uuid, String body, boolean delete, Set<Integer> allowedStatuses) throws Exception {
+        String uri = baseUrl + PATH + resource;
+        if (uuid != null) {
+            uri += ("/" + uuid);
+        }
+
+        final String userAndPass = username + ":" + new String(password);
+        byte[] auth = Base64.getEncoder().encode(userAndPass.getBytes(UTF_8));
+        HttpRequest.Builder reqBuilder =
+                HttpRequest.newBuilder().setHeader(HttpHeaders.AUTHORIZATION, "Basic " + new String(auth, UTF_8));
```

Comment:
Will this function when SSO is enabled?

Reply:
In any case, this client is a temporary addition, after the fhir2 module is modified to support the features we want, we should get rid of it.

Reply:
The MVP should work with OpenMRS HIS, which has SSO enabled. **This needs to be fixed**.

Reply:
Done